src/metabase/actions/execution.clj

```diff
@@ -177,18 +167,17 @@
   (let [dashcard (api/check-404 (db/select-one DashboardCard
                                                :id dashcard-id
                                                :dashboard_id dashboard-id))
-        action (api/check-404 (first (action/select-actions :id (:action_id dashcard))))]
-    (if (= (:type action) "implicit")
-      (let [model-action (api/check-404 (first (action/actions-with-implicit-params nil :id (:id action))))]
-        (execute-implicit-action model-action request-parameters))
+        action (api/check-404 (first (action/actions-with-implicit-params nil :id (:action_id dashcard))))]
+    (if (= :implicit (:type action))
```

Comment:
Cam told me this was Yoda speech when I wrote the condition this way. (Although we write tests like this (first comes the expected value).)

Reply:
Ah, for me it's left over java brain. `CONSTANT.equals(var)` won't throw an NPE.

Reply:
I might have the same in the subconscious.